### PR TITLE
[WIP] Enabling GraphProfilingExecutor for checkScript*

### PR DESCRIPTION
**Pull request by Krovatkin**

### CODE REVIEW

File: torch/csrc/jit/profiling_record.cpp [hunk 1]
```diff
@@ -18,7 +18,67 @@ ProfileOp* ProfilingRecord::createProfileNode(
...
+      i->setType(TensorType::get());
+    }
+  }
+}
+
+static void unprofileBlock(Block *block) {
+  for (auto it = block->nodes().begin(); it != block->nodes().end(); ++it) {
+    auto n = *it;
+    for (auto o : n->outputs()) {
+      if (o->type()->isSubtypeOf(TensorType::get())) {
```
Comment (zdevito) [inline torch/csrc/jit/profiling_record.cpp]: Probably want to use unshapedType, what happens if Tensors are an element of a tuple or list?

File: torch/csrc/jit/register_prim_ops.cpp [hunk 2]
```diff
@@ -871,7 +877,8 @@ RegisterOperators reg(
          },
          aliasAnalysisSpecialCase()),
      Operator(
-         prim::AutogradAnyNonZero,
+         //prim::AutogradAnyNonZero,
+         "prim::AutogradAnyNonZero(...) -> int",
```
Comment (zdevito) [inline torch/csrc/jit/register_prim_ops.cpp]: `-> bool`

**Review by zdevito — COMMENTED** [review]
Thanks for putting this all together. I have a bunch of comments, though very few major ones. In the future, it would better to split this up into about 4 PRs:

* changes to autodiff into propagate constants 
* Bug fixes for the loops in bailout graphs
* Improvements to guard elimination
* Updates to profiling executor to turn it on.

File: torch/csrc/jit/passes/alias_analysis.cpp [hunk 3]
```diff
@@ -349,12 +355,17 @@ void AliasDb::analyzeImpl(Node* node) {
...
+        makePointerTo(node->output(), node->inputs().at(0));
+      }
+      return;
+    case prim::BailOut:
+      TORCH_INTERNAL_ASSERT(node->inputs().at(0)->node()->kind() ==
+                            prim::BailoutTemplate);
+      makePointerTo(node->output(), node->inputs().at(1));
+      return;
+    case prim::Guard:
+      makePointerTo(node->output(), node->inputs().at(0));
```
[the remark this answers is not in the document]
Comment (suo) [reply torch/csrc/jit/passes/alias_analysis.cpp]: Not specifically—generally stuff like "point the output to *this* particular input" is done by the alias annotations in the schema. But if these ops are unschematized we can't use that.

**Pull request closed without merging**